backend/src/controllers/course.js

```diff
@@ -14,8 +14,10 @@ exports.getCourses = function (_, res, next) {
 }
 
 /* Get specifc course data */
-exports.getCourse = function ({ params }, res, next) {
-    courseModel.getCourse(params.code)
+exports.getCourse = function ({ params, user }, res, next) {
+    const userID = (user && user.id) || ANONYMOUS
+    Promise.all([courseModel.getCourse(params.code), reviewModel.userHasReviewed(params.code, userID)])
```

Comment:
Hmm danger here is I can still navigate directly to the `course/review/new` route right?

Reply:
Yes that's true, although i think it's an unlikely path.
Note that the user will still be unable to POST a new review cos it's enforced on the postReview route.
I did notice that if an error occurs on posting a review, it still tries to navigate to the new review.
Might just push a fix for that and be done with it for now.

A more foolproof fix would need to hit the course endpoint (or dedicated 'hasReviewed' endpoint?) in NewReview.